app.js

```diff
@@ -26,14 +26,22 @@ server.post('/skill', function(req, res, next){
 	if(!req.body || req.body.name===''){
 		res.send(400, {error: "please add a name to create a skill"})
 	}
-	var skill = new Skill;
-	skill._id = new mongoose.mongo.ObjectID();
-	skill.name = req.body.name;
-	skill.save(function(err){
-		if (err){
-			res.send(400, {error:err.message});
+
+	Skill.findOne({name:req.params.name}, function(err, skill){
```

Comment:
Consider using lean to return less db metadata for a faster query
https://mongoosejs.com/docs/tutorials/lean.html

`Skill.findOne({name:req.params.name}, null, {lean:true}, function(err, skill){`
should work without rewriting to an async